python/ray/utils.py

```diff
@@ -496,6 +497,141 @@ def is_main_thread():
     return threading.current_thread().getName() == "MainThread"
 
 
+def set_psigdeath(child_proc, _globals=[None]):
+    """Ensures the child process dies with the parent (fate-sharing).
+
+    On Linux, this must be called by the child in preexec_fn.
+    On Windows, this must be called by the parent, after spawning.
+    
+    Args:
+        child_proc:
+            On Linux, the child PID (os.getpid()).
+            On Windows, the subprocess.Popen or subprocess.Handle object.
+
+    Returns:
+        False if the functionality is unavailable.
+        Otherwise, returns True if the call succeeds, or if child_proc is None.
+        Raises an error if the functionality is available but the call fails.
+    """
+
+    _set_psigdeath = _globals[0]
```

Comment:
What's the purpose of this? Caching the definition for performance? Seems unnecessary

Reply:
See my reply to the other comment.